src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final double INCREASING_ARRAY = 1.5;
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] defaultArray;
+    private int size = 0;
+
+    public ArrayList() {
+        defaultArray = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        defaultArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        checkSize();
+        if (index == size) {
+            add(value);
+            return;
+        }
+        System.arraycopy(defaultArray, index, defaultArray, index + 1, size - index);
+        defaultArray[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int index = 0; index < list.size(); index++) {
+            add(list.get(index));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return defaultArray[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        defaultArray[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T value = defaultArray[index];
+        System.arraycopy(defaultArray, index + 1, defaultArray, index, size - index - 1);
+        size--;
+        return value;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int index = 0; index < size; index++) {
+            if (Objects.equals(defaultArray[index], element)) {
+                return remove(index);
+            }
+        }
+        throw new NoSuchElementException("No such element in ArrayList");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkSize() {
```

Comment:
This is a bad name. Checking is not the main goal of this method.
`private void increaseCapacityIfFull()`
or something like this